ftw/news/browser/news_listing.py

```diff
@@ -0,0 +1,179 @@
+from Acquisition import aq_parent, aq_inner
+from ftw.news import _
+from ftw.news import utils
+from ftw.news.interfaces import INewsListingView
+from plone.portlets.interfaces import IPortletAssignmentMapping
+from plone.portlets.interfaces import IPortletManager
+from plone.portlets.interfaces import IPortletRenderer
+from Products.CMFCore.permissions import AccessInactivePortalContent
+from Products.CMFCore.utils import _checkPermission
+from Products.CMFCore.utils import getToolByName
+from Products.CMFPlone.interfaces import IPloneSiteRoot
+from Products.CMFPlone.PloneBatch import Batch
+from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
+from zope.component import getMultiAdapter
+from zope.component import getUtility
+from zope.component import queryMultiAdapter
+from zope.interface import implements
+from zope.publisher.browser import BrowserView
+import DateTime
+import datetime
+
+
+class NewsListing(BrowserView):
+    implements(INewsListingView)
+    template = ViewPageTemplateFile('templates/news_listing.pt')
+
+    def __init__(self, context, request):
+        super(NewsListing, self).__init__(context, request)
+        self.batch = None
+
+    def __call__(self):
+        b_start = self.request.form.get('b_start', 0)
+        self.batch = Batch(self.get_items(), 10, b_start)
+        return self.template()
+
+    def get_items(self):
+        query = {
+            'object_provides': 'ftw.news.interfaces.INews',
+            'sort_on': 'start',
+            'sort_order': 'reverse'
+        }
+        show_inactive = _checkPermission(AccessInactivePortalContent,
+                                         self.context)
+        catalog = getToolByName(self.context, 'portal_catalog')
+        # TODO: The following line makes RSS feed only work on news folders.
+        query['path'] = '/'.join(self.context.getPhysicalPath())
+
+        datestring = self.request.get('archive')
+        if datestring:
+            try:
+                start = DateTime.DateTime(datestring)
+            except DateTime.interfaces.SyntaxError:
+                raise
+            end = DateTime.DatetTime('%s/%s/%s'.format(
```

Comment:
`DatetTime` ?
Please add a test for this section. :wink: 
You may run `bin/test-coverage` to have an overview about what is not yet tested..